File: src/model/historySession.ts
import { HistoryType } from "../util/typeUtil";

export class HistorySession {
  historyList: HistoryType[];
  constructor() {
    this.historyList = [];
  }

  getList() {
    return this.historyList;
  }

  add(history: HistoryType) {
    this.historyList = [...this.historyList, history];
  }
}

File: src/main/java/org/example/model/HistorySession.java
package org.example.model;

import java.util.ArrayList;
import java.util.List;
import org.example.utils.HistoryType;

public class HistorySession {
  private List<HistoryType> historyList;

  public HistorySession() {
    this.historyList = new ArrayList<>();
  }

  public List<HistoryType> getList() {
    return this.historyList;
  }

  public void add(HistoryType history) {
    this.historyList.add(history);
  }
}
